api/src/main/java/org/infinispan/api/common/events/KeyValueListener.java

```diff
@@ -0,0 +1,53 @@
+package org.infinispan.api.common.events;
+
+/**
+ * KeyValueListener interface to be used with caches
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ * @since 10.0
+ */
+public interface KeyValueListener<K, V> {
+
+   default boolean sync() {
+      return true;
+   }
+
+   default boolean primaryOnly() {
+      return false;
+   }
+
+   default boolean clustered() {
```

Comment:
I am debating maybe clustered and primary only and local should be an enum instead, they are all mutually exclusive.

Reply:
We could even include current state and observation in this enum since it requires being clustered for current state and observation only applies for primary or local.

Reply:
Not sure how this allows us to decide if something is clustered or not though? Publisher/Multi can be used for a local listener just the same as clustered.

Reply:
I was thinking the notification publisher/multi would be clustered by default, and local if the cache was decorated with the `CACHE_MODE_LOCAL` flag.